### PR TITLE
ethtool: drop NetDevAdvertise

### LINKED ISSUE #10807: binary sizes for systemd-udevd and udevadm exploded
building current git master, the sizes of the udevadm and systemd-udevd binary have increased over 10 times, both are over 33M in size.
```
michael@pluto:~/debian/trunk-ci/systemd-239+upstream20181116 (master)$ du -hs debian/udev/lib/systemd/systemd-udevd debian/udev/bin/udevadm 
33M	debian/udev/lib/systemd/systemd-udevd
33M	debian/udev/bin/udevadm

michael@pluto:~/debian/trunk-ci/systemd-239+upstream20181116 (master)$ file debian/udev/lib/systemd/systemd-udevd debian/udev/bin/udevadm 
debian/udev/lib/systemd/systemd-udevd: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=02d261727ce051abc0ebbccaa94d3adfb3d08292, stripped
debian/udev/bin/udevadm:               ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=981a95a0fad1a6d10d42752189e595af0d690f61, stripped
```

This seems to be a recent regression, so marking that as a blocker for v240.

### PR DESCRIPTION
**Pull request by yuwata**
Fixes #10807.

### CODE REVIEW

File: src/udev/net/ethtool-util.c [hunk 1]
```diff
@@ -733,15 +769,26 @@ int config_parse_advertise(const char *unit,
...
+                a[i] |= 1UL << j;
         }
 
-        config->advertise |= a;
+        if (!config->advertise) {
+                config->advertise = new(uint32_t, ETHTOOL_LINK_MODE_MASK_MAX_KERNEL_NU32);
+                if (!config->advertise)
+                        return log_oom();
+
+                memcpy(config->advertise, a, ETHTOOL_LINK_MODE_MASK_MAX_KERNEL_NBYTES);
```
Comment (keszybz) [inline src/udev/net/ethtool-util.c:785]: So I don't think we want to allocate this. Right now those values fit into one 64 bit number, so let's not overcomplicate things. If the kernel ever grows new modes, we can reconsider.

Comment (yuwata) [reply src/udev/net/ethtool-util.c:785]: Hmm, but 51 fields are already defined... And, I do not think this makes code so complicate.

**Comment by yuwata** [comment]
Updated.
- missing.h is slightly modified.
- add forgotten `free()` for `link_config::advertise`.

**Comment by medhefgo** [comment]
Just saw this now. You added a bunch of modes here that aren't mentioned in the docs, maybe they should be added?

**Comment by keszybz** [comment]
Ah right, we need to do that.

**Pull request merged**